File: src/services/tauri/dxt.ts
import { getJson, invokeTauri, isDesktopTauri, postJson, postNoContent } from './shared';

export async function loadManifests() {
  if (isDesktopTauri()) {
    return await invokeTauri<unknown[]>('load_manifests');
  }
  return await getJson<unknown[]>('/api/dxt/manifests');
}

export async function loadManifest(user: string, repo: string) {
  if (isDesktopTauri()) {
    return await invokeTauri<unknown | null>('load_manifest', { user, repo });
  }
  return await postJson<unknown | null>('/api/dxt/manifest', { user, repo });
}

export async function checkManifestsExist() {
  if (isDesktopTauri()) {
    return await invokeTauri<boolean>('check_manifests_exist');
  }
  return await getJson<boolean>('/api/dxt/manifests/exist');
}

export async function downloadAndExtractManifests() {
  if (isDesktopTauri()) {
    await invokeTauri('download_and_extract_manifests');
    return;
  }
  await postNoContent('/api/dxt/manifests/download');
}

export async function readDxtSetting(user: string, repo: string) {
  if (isDesktopTauri()) {
    return await invokeTauri<unknown>('read_dxt_setting', { user, repo });
  }
  return await postJson<unknown>('/api/dxt/setting/read', { user, repo });
}

export async function saveDxtSetting(user: string, repo: string, content: unknown) {
  if (isDesktopTauri()) {
    await invokeTauri('save_dxt_setting', { user, repo, content });
    return;
  }
  await postNoContent('/api/dxt/setting/save', { user, repo, content });
}

File: src-tauri/src/commands/dxt.rs
use serde_json::Value;

#[tauri::command]
pub async fn load_manifests() -> Result<Value, String> {
    codexia_shared::dxt::load_manifests().await
}

#[tauri::command]
pub async fn load_manifest(user: String, repo: String) -> Result<Value, String> {
    codexia_shared::dxt::load_manifest(user, repo).await
}

#[tauri::command]
pub async fn read_dxt_setting(user: String, repo: String) -> Result<Value, String> {
    codexia_shared::dxt::read_dxt_setting(user, repo).await
}

#[tauri::command]
pub async fn save_dxt_setting(user: String, repo: String, content: Value) -> Result<(), String> {
    codexia_shared::dxt::save_dxt_setting(user, repo, content).await
}

#[tauri::command]
pub async fn download_and_extract_manifests() -> Result<(), String> {
    codexia_shared::dxt::download_and_extract_manifests().await
}

#[tauri::command]
pub async fn check_manifests_exist() -> Result<bool, String> {
    codexia_shared::dxt::check_manifests_exist().await
}
